rs/ic_os/config/src/update_config.rs

```diff
@@ -0,0 +1,532 @@
+                    address: "2001:db8::1/64".to_string(),
+                    gateway: "2001:db8::fffe".parse().unwrap(),
+                }),
+                ipv4_config: Some(Ipv4Config {
+                    address: "192.0.2.1".parse().unwrap(),
+                    prefix_length: 24,
+                    gateway: "192.0.2.254".parse().unwrap(),
+                }),
+                domain_name: Some("example.com".to_string()),
+            }
+        );
+
+        assert_eq!(result.hostname, Some("guest-001122334455".to_string()));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_read_filebeat_conf_existing_file() -> Result<()> {
+        let dir = tempdir()?;
+        let filebeat_conf_path = dir.path().join("filebeat.conf");
+        let mut file = fs::File::create(&filebeat_conf_path)?;
+        writeln!(file, "elasticsearch_hosts=host1:9200,host2:9200")?;
+        writeln!(file, "elasticsearch_tags=tag1,tag2")?;
+
+        let logging = read_filebeat_conf(dir.path())?;
+
+        assert_eq!(
+            logging.elasticsearch_hosts,
+            "host1:9200,host2:9200".to_string()
+        );
+        assert_eq!(logging.elasticsearch_tags, Some("tag1,tag2".to_string()));
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_read_filebeat_conf_missing_file() -> Result<()> {
+        let dir = tempdir()?;
+        let logging = read_filebeat_conf(dir.path())?;
+
+        assert_eq!(
+            logging.elasticsearch_hosts,
+            "elasticsearch-node-0.mercury.dfinity.systems:443 \
+            elasticsearch-node-1.mercury.dfinity.systems:443 \
+            elasticsearch-node-2.mercury.dfinity.systems:443 \
+            elasticsearch-node-3.mercury.dfinity.systems:443"
+                .to_string()
+        );
+        assert_eq!(logging.elasticsearch_tags, None);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_read_nns_conf_existing_file() -> Result<()> {
+        let dir = tempdir()?;
+        let nns_conf_path = dir.path().join("nns.conf");
+        let mut file = fs::File::create(&nns_conf_path)?;
+        writeln!(
+            file,
+            "nns_url=https://nns1.example.com,https://nns2.example.com"
+        )?;
+
+        let nns_urls = read_nns_conf(dir.path())?;
+
+        assert_eq!(
+            nns_urls,
+            vec![
+                Url::parse("https://nns1.example.com")?,
+                Url::parse("https://nns2.example.com")?,
+            ]
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_read_nns_conf_missing_file() -> Result<()> {
+        let dir = tempdir()?;
+        let nns_urls = read_nns_conf(dir.path())?;
+
+        assert_eq!(
+            nns_urls,
+            vec![
+                Url::parse("https://icp-api.io")?,
+                Url::parse("https://icp0.io")?,
+                Url::parse("https://ic0.app")?,
+            ]
+        );
+
+        Ok(())
+    }
+}
+
+// "DOWNGRADE" guestOS config logic
+//
```

Comment:
Update: I've update the config revamp integration to continue passing the old GuestOS config, so we no longer need the GuestOS "downgrade-config" logic. I've commented it out.